### PR TITLE
feature/atlassian-workflow

### PR DESCRIPTION
**Pull request by Benimanela**

### CODE REVIEW

File: atlassian/atlassian-workflow.xml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+        <!-- Prepare the lastEventsTime path -->
+        <Set path="/lastEventsTime" value="[]"/>
+
+        <!-- Merge the log data from the API response into the "/lastEventsTime" path. -->
+        <Merge targetPath="/lastEventsTime" sourcePath="/getEvents/body/data/attributes/time" />
+
+        <!-- Set the number of events -->
+        <Set path="/currentFetchEvents" value="${count(/getEvents/body/data/id)}"/>
+        <Set path="/numberOfEvents" value="${count(/getEvents/body/data/id)}"/>
```
Comment (LiorQM) [inline atlassian/atlassian-workflow.xml:53]:
```suggestion
        <!-- Set the number of events -->
        <Set path="/currentFetchEvents" value="${count(/getEvents/body/data/id)}"/>
        <Set path="/numberOfEvents" value="${/currentFetchEvents}"/>

        <!-- Post Events, if any -->
        <If condition="/currentFetchEvents > 0">
            <PostEvents path="/getEvents/body/data" source="${/orgID}" />
        </If>

        <!-- Prepare the lastEventsTime path -->
        <Set path="/lastEventsTime" value="[]"/>

        <!-- Merge the log data from the API response into the "/lastEventsTime" path. -->
        <Merge targetPath="/lastEventsTime" sourcePath="/getEvents/body/data/attributes/time" />
```
You are using here the `count()` function 3 times:
1. For checking whether any data has returned.
2. To get the number of fetched events.
3. To initialize the total number of fetched events.

File: atlassian/atlassian-workflow.xml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            </If>
+
+            <!-- Post Events, if any -->
+            <If condition="count(/getEvents/body/data) > 0">
+                <PostEvents path="/getEvents/body/data" source="${/orgID}" />
+            </If>
+
+            <!-- Set the number of events -->
+            <Set path="/currentFetchEvents" value="${count(/getEvents/body/data/id)}"/>
+            <Set path="/numberOfEvents" value="${/numberOfEvents + /currentFetchEvents}"/>
```
Comment (LiorQM) [inline atlassian/atlassian-workflow.xml:80]:
```suggestion
            <!-- Set the number of events -->
            <Set path="/currentFetchEvents" value="${count(/getEvents/body/data/id)}"/>
            <Set path="/numberOfEvents" value="${/numberOfEvents + /currentFetchEvents}"/>

            <!-- Post Events, if any -->
            <If condition="count(/getEvents/body/data) > 0">
                <PostEvents path="/getEvents/body/data" source="${/orgID}" />
            </If>
```

File: atlassian/atlassian-workflow.xml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            <!-- Extract the next link for the next iteration -->
+            <Set path="/nextLink" value="${/getEvents/body/links/next}" />
+        </While>
+
+        <Log type="INFO" message="Total number of fetched events: ${/numberOfEvents}."/>
+
+        <If condition="count(/lastEventsTime) > 0">
+            <!-- Update the bookmark -->
+            <Set path="/lastEvent" value="${max(/lastEventsTime)}" />
+            <ParseDate pattern="yyyy-MM-dd'T'HH:mm:ss[.SSS]'Z'" timeZone="${/timeZone}" date="${max(/lastEventsTime)}" savePath="/lastEventTime" />
```
Comment (LiorQM) [inline atlassian/atlassian-workflow.xml:98]: You've set `lastEvent`, but haven't used it and instead ran `max()` twice.

File: atlassian/atlassian-workflow.xml (hunk 1 above)
Comment (Benimanela) [reply atlassian/atlassian-workflow.xml:53]: fixed

File: atlassian/atlassian-workflow.xml (hunk 3 above)
Comment (Benimanela) [reply atlassian/atlassian-workflow.xml:98]: fixed

File: atlassian/atlassian-workflow.xml [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+            <PostEvents path="/getEvents/body/data" source="${/orgID}" />
+        </If>
+
+        <!-- Prepare the lastEventsTime path -->
+        <Set path="/lastEventsTime" value="[]"/>
+
+        <!-- Merge the log data from the API response into the "/lastEventsTime" path. -->
+        <Merge targetPath="/lastEventsTime" sourcePath="/getEvents/body/data/attributes/time" />
+
+        <Log type="INFO" message="Page number: ${/pageNumber}, Number of events in this fetched: ${/currentFetchEvents}"/>
```
Comment (LiorQM) [inline atlassian/atlassian-workflow.xml:55]:
```suggestion
        <Log type="INFO" message="Number of events fetched: ${/currentFetchEvents}"/>
```

File: atlassian/atlassian-workflow.xml [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+            <Set path="/numberOfEvents" value="${/numberOfEvents + /currentFetchEvents}"/>
+
+            <!-- Post Events, if any -->
+            <If condition="count(/currentFetchEvents) > 0">
+                <PostEvents path="/getEvents/body/data" source="${/orgID}" />
+            </If>
+
+            <!-- Set the next page number -->
+            <Set path="/pageNumber" value="${/pageNumber + 1}"/>
+            <Log type="INFO" message="Page number: ${/pageNumber}, Number of events in this fetched: ${/currentFetchEvents}"/>
```
Comment (LiorQM) [inline atlassian/atlassian-workflow.xml:85]:
```suggestion
            <Log type="INFO" message="Number of events fetched: ${/currentFetchEvents}"/>
```

File: atlassian/atlassian-workflow.xml [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+        <!-- Initialize the risingValue filter for the first run. -->
+        <If condition="count(/startTime) > 0" >
+            <Initialize path="/bookmark" value="${/startTime}"/>
+        </If>
+        <Else>
+            <Initialize path="/bookmark" value="${time() - 3600000}" />
+        </Else>
+
+        <!-- Set the firse page number -->
+        <Set path="/pageNumber" value="1"/>
```
Comment (LiorQM) [inline atlassian/atlassian-workflow.xml:23]:
```suggestion
```

**Review by LiorQM — APPROVED** [review]
Page number isn't needed, other than that all good.

File: atlassian/atlassian-workflow.xml (hunk 4 above)
Comment (Benimanela) [reply atlassian/atlassian-workflow.xml:55]: why not, it help to debugging

**Pull request merged**